src/engine/strat_engine/pool.rs

```diff
@@ -363,15 +406,161 @@ impl Recordable<PoolSave> for StratPool {
+                                                    .iter()
+                                                    .map(&lookup)
+                                                    .collect());
+
+    let thin_data_segments: Vec<Segment> = try!(pool_save
+                                                    .flex_devs
+                                                    .thin_data_dev
+                                                    .iter()
+                                                    .map(&lookup)
+                                                    .collect());
+
+    let mut thin_meta_spare_segments: Vec<Segment> = try!(pool_save
+                                                              .flex_devs
+                                                              .thin_meta_dev_spare
+                                                              .iter()
+                                                              .map(&lookup)
+                                                              .collect());
+
+    let dm = try!(DM::new());
+
+    let meta_name = format_flex_name(&pool_uuid, FlexRole::ThinMeta);
+
+    // The cleanup zone.
+    let meta_dev = try!(LinearDev::new(&meta_name, &dm, thin_meta_segments));
+
+    let data_dev = try!(LinearDev::new(&format_flex_name(&pool_uuid, FlexRole::ThinData),
+                                       &dm,
+                                       thin_data_segments));
+
+    let thinpool_name = format_thinpool_name(&pool_uuid, ThinPoolRole::Pool);
+    let thinpool_dev = match ThinPoolDev::setup(&thinpool_name,
+                                                &dm,
+                                                try!(data_dev.size()),
+                                                pool_save.thinpool_dev.data_block_size,
+                                                DATA_LOWATER,
+                                                meta_dev,
+                                                data_dev) {
+        Ok(dev) => dev,
+        Err(DmError::Dm(devicemapper::ErrorEnum::CheckFailed(meta_dev, data_dev), _)) => {
+            let mut spare_dev = try!(LinearDev::new(&format_flex_name(&pool_uuid,
+                                                                      FlexRole::ThinMetaSpare),
+                                                    &dm,
+                                                    thin_meta_spare_segments.drain(..).collect()));
+
+            if try!(Command::new("thin_repair")
+                        .arg("-i")
+                        .arg(&try!(meta_dev.devnode()))
+                        .arg("-o")
+                        .arg(&try!(spare_dev.devnode()))
+                        .status())
+                       .success() == false {
+                return Err(EngineError::Engine(ErrorEnum::Error,
+                                               "thin_repair failed, pool unusable".into()));
+            }
+
```

Comment:
there was also a comment here. I really do think a comment here or in the function's comments isn't excessive, so that the meta/meta_spare switch is explained a little.